### DISCUSSION TITLE
Implementing "Union Type" for specific string values

### DISCUSSION THREAD

**Discussion by ABNERMATHEUS** [dotnet/csharplang#8459] [Language Ideas]
Issue: Implement Implementing "Union Type" for specific string values without using Enum
Description:
In TypeScript, it is possible to define a set of specific values for a variable using union types, as shown in the example below:

`type Status = 'pending' | 'received';
`

This approach is straightforward, avoids the creation of an enum, and can be used directly in string-type variables, keeping the code clean.

Currently, in C#, the common solution for this scenario is the creation of enums. However, creating an enum can sometimes be excessive when only a small number of fixed string values is needed. Additionally, enums restrict the flexibility of using string values directly.

Proposal:
Implement a simple way to define fixed string values without the need for enums to restrict the possible values that a string can hold.

`    public 'received' | 'pending' | 'progress' Status { get; set; }
`

**Comment by HaloFour** [comment]
See: #2849

**Comment by balakhaniyan** [comment]
I think using Enum is much more better than string like `Python` and `TypeScript`, but it can be more concise like `record`:
```C#
enum Status(Pending, Received, Prgoress);
```
Also if we can use string as Enum values, it will be more usefull:
```C#
enum Status {
  Pending = "pending",
  Received = "received",
  Progress = "progress"
}
```
and in concise way:

```C#
enum Status(Pending = "pending", Received = "received", Prgoress = "progress");
```

**Reply by FaustVX** [reply]
Your consice way:
```C#
enum Status(Pending, Received, Prgoress);
```
is longer (by a single character) than a regular `enum`, just replace `()` with `{}` and remove `;` 😄 
```C#
enum Status{Pending, Received, Prgoress}
```

**Reply by colejohnson66** [reply]
String versions already exist. Just call `ToString()` on your value.

**Reply by balakhaniyan** [reply]
@colejohnson66 
I don't think so, your way is OK if key and value be same, but for `Pending = "Please Wait"` it is not correct.

**Reply by balakhaniyan** [reply]
@FaustVX 
Yes, but you are breaking C# clean code rules :)... curly braces and enum values must be single in their lines...
